modules/govuk_jenkins/manifests/ssh_slave.pp

```diff
@@ -0,0 +1,60 @@
+# == Define: Govuk_jenkins::Ssh_slave
+#
+# Create an SSH slave that the Jenkins master can use to build jobs
+#
+# === Parameters:
+#
+# [*agent_hostname*]
+#   The hostname of the agent that the master connects to.
+#
+# [*credentials_id*]
+#   This is a uniquely generated ID that is created when you create a set of
+#   credentials. The ID is linked to the specific Jenkins instance, so must
+#   be created manually before being added to this defined type.
+#
+# [*description*]
+#   Describe what the agent is used for.
+#
+# [*executors*]
+#   Set how many jobs the instance can build at any one time.
+#
+# [*ssh_port*]
+#   Set the SSH port that the master connects to.
+#
+# [*jenkins_home*]
+#   Home directory of the Jenkins user, which also sets the root directory of
+#   where builds take place.
+#
+# [*jenkins_user*]
+#   The user which the master connects to the agent over SSH as.
+#
+define govuk_jenkins::ssh_slave (
+  $agent_hostname,
+  $credentials_id,
+  $labels = undef,
+  $description = 'Generic build agent',
+  $executors = '4',
```

Comment:
I did it to annoy you ;) I'll change them. There's no reason other than absent mindedness.